app/components/school-session-type-form.js

```diff
@@ -19,22 +20,57 @@ const Validations = buildValidations({
       regex: /^#[a-fA-F0-9]{6}$/,
     })
   ],
+  selectedAamcMethodId: [
+    validator(function(aamcMethodId, options, model) {
+      if (!aamcMethodId) {
+        return true;
+      }
+      const assessment = model.get('assessment');
+      const lookup = assessment?'AM':'IM';
+      if (aamcMethodId.indexOf(lookup) !== 0) {
+        return this.createErrorMessage(options.messageKey, aamcMethodId, options);
+      }
+
+      return true;
+    }, {
+      dependentKeys: ['model.assessment'],
+      descriptionKey: 'general.aamcMethod',
+      messageKey: 'general.invalid',
+    })
+  ]
 });
 
 export default Component.extend(ValidationErrorDisplay, Validations, {
   store: service(),
   classNames: ['school-session-type-form'],
   title: null,
+  selectedAamcMethodId: null,
   calendarColor: null,
   assessment: null,
   assessmentOptionId: null,
+  aamcMethods: computed('assessment', async function(){
```

Comment:
Performance can be improved here by splitting this into an allAaamcMethods property and a filtered aamcMethids property.